bin/combine_spectra

```diff
@@ -415,6 +396,53 @@ def fix_arf_exposure( arfs, outarf, method ):
     tab.write()
 
 
+def fix_rmf_numkeywords(outrmf):
+    """
+    The NUMGRP and NUMELT keywords should be equal to the sum of 
+    the N_GRP and N_CHAN columns, respectively.   It looks like 
+    instead they are just copied from the 1st input file.
+    
+    Go ahead an update them.    
+    """
+    
+    if "" == outrmf:
+        return
+    
+    # Since RMF files are special: variable length columns and 
+    # multiple extensions (EBOUNDS); I don't trust that crates will 
+    # open in rw mode will work correctly.  So I'm going to use
+    # dmhedit to just modify keywords in-place.
+
+    from pycrates import read_file
+    tab = read_file(outrmf+"[cols n_grp,n_chan]", mode="r")
+    chan = tab.get_column("n_chan").values
+    numelt = sum( [sum(c) for c in chan]) # variable length array col
+    verb2("The updated NUMELT keyword is {}".format(numelt))
+
+    grp = tab.get_column("n_grp").values
+    numgrp = sum(grp)
+    verb2("The updated NUMGRP keyword is {}".format(numgrp))
+    
+    from ciao_contrib.runtool import dmhedit
+    dmhedit.infile=outrmf
+    dmhedit.file=""
```

Comment:
We *could* avoid calling `dmhedit` twice and use the `filelist` option here (simplify the `dmhistory` output slightly, but at the expense of losing information on what has changed). I thought I had a routine to simplify this (i.e. generating the temporary file), but it turns out I don't. So I don't think it's a worthwhile change. But we now have proof that I decided against suggesting this!